File: jphp-game-ext/src/main/java/org/develnext/jphp/ext/game/classes/UXGameEntity.java
package org.develnext.jphp.ext.game.classes;

import javafx.scene.Node;
import org.develnext.jphp.ext.game.GameExtension;
import org.develnext.jphp.ext.game.support.GameEntity2D;
import org.develnext.jphp.ext.game.support.Vec2d;
import php.runtime.annotation.Reflection.*;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

@Namespace(GameExtension.NS)
public class UXGameEntity extends BaseWrapper<GameEntity2D> {
    interface WrappedInterface {
        @Property GameEntity2D.BodyType bodyType();
        @Property @Nullable Vec2d gravity();

        @Property double gravityX();
        @Property double gravityY();

        @Property double velocityX();
        @Property double velocityY();

        @Property Vec2d velocity();

        @Property Vec2d angleSpeed();
        @Property double speed();
        @Property double direction();
    }

    public UXGameEntity(Environment env, GameEntity2D wrappedObject) {
        super(env, wrappedObject);
    }

    public UXGameEntity(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct(String entityType, Node node) {
        __wrappedObject = new GameEntity2D(entityType, node);
    }

    @Getter
    public UXGameScene getGameScene(Environment env) {
        return getWrappedObject().getScene() == null ? null : new UXGameScene(env, getWrappedObject().getScene());
    }
}

File: jphp-game-ext/src/main/resources/JPHP-INF/sdk/php/game/UXGameEntity.php
<?php
namespace php\game;

use php\gui\UXNode;
use php\gui\UXParent;

/**
 * Class UXGameObject
 * @package php\game
 */
class UXGameEntity
{
    /**
     * @var UXNode
     */
    public $node;

    /**
     * @var float
     */
    public $x;

    /**
     * @var float
     */
    public $y;

    /**
     * @var float
     */
    public $centerX;

    /**
     * @var float
     */
    public $centerY;

    /**
     * @var string STATIC, DYNAMIC, KINEMATIC
     */
    public $bodyType = 'STATIC';

    /**
     * If null, using scene gravity
     * @var array|null [x, y]
     */
    public $gravity = null;

    /**
     * @var float
     */
    public $gravityX = 0.0;

    /**
     * @var float
     */
    public $gravityY = 0.0;

    /**
     * @readonly
     * @var UXGameScene
     */
    public $gameScene = null;

    /**
     * @var array
     */
    public $velocity = [0, 0];

    /**
     * @var float
     */
    public $velocityX = 0.0;

    /**
     * @var float
     */
    public $velocityY = 0.0;

    /**
     * @var array alias of velocity property
     */
    public $angleSpeed = [0, 0];

    /**
     * @var float angle speed
     */
    public $speed = 0.0;

    /**
     * @var float angle for speed, from 0 to 360
     */
    public $direction = 0.0;

    /**
     * @var float alias of velocityX
     */
    public $hspeed = 0.0;

    /**
     * @var float alias of velocityY
     */
    public $vspeed = 0.0;

    /**
     * @param string $entityType
     * @param UXNode $node
     */
    public function __construct($entityType, UXNode $node)
    {
    }

    /**
     * @param string $entityType
     * @param callable|null $handler
     */
    public function setCollisionHandler($entityType, callable $handler)
    {
    }
}